In [1]:
%load_ext autoreload

In [1]:
%run -i 'main.py' --gpu_use=1 --which_gpu=0

Namespace(gpu_use=1, which_gpu=0)
['reggae', 'classical', 'country', 'jazz', 'metal', 'pop', 'disco', 'hiphop', 'rock', 'blues']
(443, 128, 1287)
(443, 128, 1287) (443,) (197, 128, 1287) (197,) (290, 128, 1287) (290,)
model_1DCNN(
  (conv0): Sequential(
    (0): Conv1d(128, 32, kernel_size=(8,), stride=(1,))
    (1): BatchNorm1d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (2): ReLU()
    (3): MaxPool1d(kernel_size=8, stride=8, padding=0, dilation=1, ceil_mode=False)
  )
  (conv1): Sequential(
    (0): Conv1d(32, 32, kernel_size=(8,), stride=(1,))
    (1): BatchNorm1d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (2): ReLU()
    (3): MaxPool1d(kernel_size=8, stride=8, padding=0, dilation=1, ceil_mode=False)
  )
  (conv2): Sequential(
    (0): Conv1d(32, 64, kernel_size=(4,), stride=(1,))
    (1): BatchNorm1d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (2): ReLU()
    (3): MaxPool1d(kernel_size=4, stride=4,

/home/jonas/git/gct634-2018/hw2/refactored_code/train.py:46: UserWarning: invalid index of a 0-dim tensor. This will be an error in PyTorch 0.5. Use tensor.item() to convert a 0-dim tensor to a Python number
  print ("Epoch [%d/%d], Iter [%d/%d] loss : %.4f" % (epoch+1, num_epochs, i+1, len(train_loader), loss.data[0]))



torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
Epoch [1/50], Iter [40/88] loss : 1.4709
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32

/home/jonas/git/gct634-2018/hw2/refactored_code/train.py:77: UserWarning: invalid index of a 0-dim tensor. This will be an error in PyTorch 0.5. Use tensor.item() to convert a 0-dim tensor to a Python number
  eval_loss += loss.data[0]


torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size

torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
Epoch [2/50], Iter [80/88] loss : 1.2809
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 3

torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
Epoch [3/50], Iter [60/88] loss : 1.7027
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 3

torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
Epoch [4/50], Iter [40/88] loss : 0.6005
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 3

torch.Size([5, 64, 4])
Epoch [5/50], Iter [10/88] loss : 1.2644
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32, 160])
torch.Size([5, 32, 19])
torch.Size([5, 64, 4])
torch.Size([5, 128, 1287])
torch.Size([5, 32,

KeyboardInterrupt: 

In [1]:
%run -i 'main.py' --gpu_use=1 --which_gpu=0

Namespace(gpu_use=1, which_gpu=0)
['reggae', 'classical', 'country', 'jazz', 'metal', 'pop', 'disco', 'hiphop', 'rock', 'blues']
(4430, 128, 128)
(4430, 128, 128) (4430,) (1970, 128, 128) (1970,) (2900, 128, 128) (2900,)
model_1DCNN_short(
  (conv0): Sequential(
    (0): Conv1d(128, 32, kernel_size=(9,), stride=(1,))
    (1): BatchNorm1d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (2): ReLU()
    (3): MaxPool1d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
  )
  (conv1): Sequential(
    (0): Conv1d(32, 32, kernel_size=(8,), stride=(1,))
    (1): BatchNorm1d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (2): ReLU()
    (3): MaxPool1d(kernel_size=3, stride=2, padding=0, dilation=1, ceil_mode=False)
  )
  (conv2): Sequential(
    (0): Conv1d(32, 64, kernel_size=(4,), stride=(3,))
    (1): BatchNorm1d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (2): ReLU()
    (3): MaxPool1d(kernel_size=2, 

/home/jonas/git/gct634-2018/hw2/refactored_code/train.py:46: UserWarning: invalid index of a 0-dim tensor. This will be an error in PyTorch 0.5. Use tensor.item() to convert a 0-dim tensor to a Python number
  print ("Epoch [%d/%d], Iter [%d/%d] loss : %.4f" % (epoch+1, num_epochs, i+1, len(train_loader), loss.data[0]))


Epoch [1/50], Iter [10/886] loss : 2.1347
Epoch [1/50], Iter [20/886] loss : 2.1169
Epoch [1/50], Iter [30/886] loss : 2.2188
Epoch [1/50], Iter [40/886] loss : 2.1656
Epoch [1/50], Iter [50/886] loss : 2.4172
Epoch [1/50], Iter [60/886] loss : 2.3318
Epoch [1/50], Iter [70/886] loss : 1.6704
Epoch [1/50], Iter [80/886] loss : 1.9135
Epoch [1/50], Iter [90/886] loss : 1.8643
Epoch [1/50], Iter [100/886] loss : 1.7372
Epoch [1/50], Iter [110/886] loss : 2.3322
Epoch [1/50], Iter [120/886] loss : 1.9622
Epoch [1/50], Iter [130/886] loss : 1.7100
Epoch [1/50], Iter [140/886] loss : 1.8688
Epoch [1/50], Iter [150/886] loss : 1.0383
Epoch [1/50], Iter [160/886] loss : 1.7515
Epoch [1/50], Iter [170/886] loss : 1.4125
Epoch [1/50], Iter [180/886] loss : 1.2602
Epoch [1/50], Iter [190/886] loss : 2.8883
Epoch [1/50], Iter [200/886] loss : 2.5271
Epoch [1/50], Iter [210/886] loss : 1.3784
Epoch [1/50], Iter [220/886] loss : 1.5482
Epoch [1/50], Iter [230/886] loss : 1.7379
Epoch [1/50], Iter [

/home/jonas/git/gct634-2018/hw2/refactored_code/train.py:77: UserWarning: invalid index of a 0-dim tensor. This will be an error in PyTorch 0.5. Use tensor.item() to convert a 0-dim tensor to a Python number
  eval_loss += loss.data[0]


Average loss: 1.7734 

Learning rate : 0.01
Epoch [2/50], Iter [10/886] loss : 1.2514
Epoch [2/50], Iter [20/886] loss : 2.1280
Epoch [2/50], Iter [30/886] loss : 1.7525
Epoch [2/50], Iter [40/886] loss : 1.2530
Epoch [2/50], Iter [50/886] loss : 2.4557
Epoch [2/50], Iter [60/886] loss : 1.6769
Epoch [2/50], Iter [70/886] loss : 1.7862
Epoch [2/50], Iter [80/886] loss : 2.4988
Epoch [2/50], Iter [90/886] loss : 1.1379
Epoch [2/50], Iter [100/886] loss : 1.4875
Epoch [2/50], Iter [110/886] loss : 1.4724
Epoch [2/50], Iter [120/886] loss : 1.5392
Epoch [2/50], Iter [130/886] loss : 1.8641
Epoch [2/50], Iter [140/886] loss : 0.9885
Epoch [2/50], Iter [150/886] loss : 1.6817
Epoch [2/50], Iter [160/886] loss : 0.7427
Epoch [2/50], Iter [170/886] loss : 1.4778
Epoch [2/50], Iter [180/886] loss : 1.5399
Epoch [2/50], Iter [190/886] loss : 1.4333
Epoch [2/50], Iter [200/886] loss : 1.2809
Epoch [2/50], Iter [210/886] loss : 1.7117
Epoch [2/50], Iter [220/886] loss : 2.8890
Epoch [2/50], Iter 

Epoch [4/50], Iter [180/886] loss : 1.1655
Epoch [4/50], Iter [190/886] loss : 0.6103
Epoch [4/50], Iter [200/886] loss : 2.1486
Epoch [4/50], Iter [210/886] loss : 1.3166
Epoch [4/50], Iter [220/886] loss : 1.3641
Epoch [4/50], Iter [230/886] loss : 0.4652
Epoch [4/50], Iter [240/886] loss : 1.5149
Epoch [4/50], Iter [250/886] loss : 1.5773
Epoch [4/50], Iter [260/886] loss : 1.3592
Epoch [4/50], Iter [270/886] loss : 0.7493
Epoch [4/50], Iter [280/886] loss : 0.8902
Epoch [4/50], Iter [290/886] loss : 1.0961
Epoch [4/50], Iter [300/886] loss : 1.7617
Epoch [4/50], Iter [310/886] loss : 0.5664
Epoch [4/50], Iter [320/886] loss : 1.4211
Epoch [4/50], Iter [330/886] loss : 1.5803
Epoch [4/50], Iter [340/886] loss : 0.4395
Epoch [4/50], Iter [350/886] loss : 1.2321
Epoch [4/50], Iter [360/886] loss : 3.1506
Epoch [4/50], Iter [370/886] loss : 1.8109
Epoch [4/50], Iter [380/886] loss : 0.7167
Epoch [4/50], Iter [390/886] loss : 0.6661
Epoch [4/50], Iter [400/886] loss : 1.8744
Epoch [4/50

Epoch [6/50], Iter [360/886] loss : 0.5628
Epoch [6/50], Iter [370/886] loss : 1.6768
Epoch [6/50], Iter [380/886] loss : 0.4028
Epoch [6/50], Iter [390/886] loss : 2.1308
Epoch [6/50], Iter [400/886] loss : 0.7520
Epoch [6/50], Iter [410/886] loss : 1.7957
Epoch [6/50], Iter [420/886] loss : 0.6067
Epoch [6/50], Iter [430/886] loss : 0.9780
Epoch [6/50], Iter [440/886] loss : 1.6630
Epoch [6/50], Iter [450/886] loss : 1.2212
Epoch [6/50], Iter [460/886] loss : 1.2454
Epoch [6/50], Iter [470/886] loss : 0.5842
Epoch [6/50], Iter [480/886] loss : 2.0614
Epoch [6/50], Iter [490/886] loss : 0.9523
Epoch [6/50], Iter [500/886] loss : 0.5305
Epoch [6/50], Iter [510/886] loss : 0.5708
Epoch [6/50], Iter [520/886] loss : 0.5602
Epoch [6/50], Iter [530/886] loss : 1.5278
Epoch [6/50], Iter [540/886] loss : 0.8080
Epoch [6/50], Iter [550/886] loss : 2.1395
Epoch [6/50], Iter [560/886] loss : 1.0505
Epoch [6/50], Iter [570/886] loss : 1.3111
Epoch [6/50], Iter [580/886] loss : 1.1054
Epoch [6/50

Epoch [8/50], Iter [480/886] loss : 0.6145
Epoch [8/50], Iter [490/886] loss : 0.3304
Epoch [8/50], Iter [500/886] loss : 0.8268
Epoch [8/50], Iter [510/886] loss : 0.2040
Epoch [8/50], Iter [520/886] loss : 0.5448
Epoch [8/50], Iter [530/886] loss : 0.3457
Epoch [8/50], Iter [540/886] loss : 0.7581
Epoch [8/50], Iter [550/886] loss : 1.3524
Epoch [8/50], Iter [560/886] loss : 1.2466
Epoch [8/50], Iter [570/886] loss : 0.5675
Epoch [8/50], Iter [580/886] loss : 1.2344
Epoch [8/50], Iter [590/886] loss : 0.3096
Epoch [8/50], Iter [600/886] loss : 1.0090
Epoch [8/50], Iter [610/886] loss : 0.6576
Epoch [8/50], Iter [620/886] loss : 0.4974
Epoch [8/50], Iter [630/886] loss : 0.6648
Epoch [8/50], Iter [640/886] loss : 2.1146
Epoch [8/50], Iter [650/886] loss : 0.4093
Epoch [8/50], Iter [660/886] loss : 0.4069
Epoch [8/50], Iter [670/886] loss : 0.9947
Epoch [8/50], Iter [680/886] loss : 0.9394
Epoch [8/50], Iter [690/886] loss : 1.7610
Epoch [8/50], Iter [700/886] loss : 1.2747
Epoch [8/50

Epoch [10/50], Iter [600/886] loss : 2.4965
Epoch [10/50], Iter [610/886] loss : 0.1749
Epoch [10/50], Iter [620/886] loss : 1.0335
Epoch [10/50], Iter [630/886] loss : 1.0411
Epoch [10/50], Iter [640/886] loss : 0.1045
Epoch [10/50], Iter [650/886] loss : 0.4673
Epoch [10/50], Iter [660/886] loss : 0.4766
Epoch [10/50], Iter [670/886] loss : 0.4828
Epoch [10/50], Iter [680/886] loss : 0.8729
Epoch [10/50], Iter [690/886] loss : 2.0038
Epoch [10/50], Iter [700/886] loss : 0.3770
Epoch [10/50], Iter [710/886] loss : 0.7973
Epoch [10/50], Iter [720/886] loss : 0.3215
Epoch [10/50], Iter [730/886] loss : 0.2632
Epoch [10/50], Iter [740/886] loss : 0.7387
Epoch [10/50], Iter [750/886] loss : 0.0098
Epoch [10/50], Iter [760/886] loss : 0.0618
Epoch [10/50], Iter [770/886] loss : 1.2020
Epoch [10/50], Iter [780/886] loss : 0.1537
Epoch [10/50], Iter [790/886] loss : 0.1748
Epoch [10/50], Iter [800/886] loss : 1.4078
Epoch [10/50], Iter [810/886] loss : 1.0795
Epoch [10/50], Iter [820/886] lo

Epoch [12/50], Iter [720/886] loss : 0.1775
Epoch [12/50], Iter [730/886] loss : 0.3535
Epoch [12/50], Iter [740/886] loss : 0.3486
Epoch [12/50], Iter [750/886] loss : 0.4399
Epoch [12/50], Iter [760/886] loss : 0.1899
Epoch [12/50], Iter [770/886] loss : 0.8569
Epoch [12/50], Iter [780/886] loss : 0.7067
Epoch [12/50], Iter [790/886] loss : 0.0556
Epoch [12/50], Iter [800/886] loss : 1.1935
Epoch [12/50], Iter [810/886] loss : 0.1477
Epoch [12/50], Iter [820/886] loss : 0.2642
Epoch [12/50], Iter [830/886] loss : 0.4334
Epoch [12/50], Iter [840/886] loss : 0.1680
Epoch [12/50], Iter [850/886] loss : 0.9127
Epoch [12/50], Iter [860/886] loss : 0.7197
Epoch [12/50], Iter [870/886] loss : 0.6713
Epoch [12/50], Iter [880/886] loss : 0.5600
Average loss: 1.9121 

Learning rate : 0.0004
Epoch [13/50], Iter [10/886] loss : 0.0998
Epoch [13/50], Iter [20/886] loss : 1.4034
Epoch [13/50], Iter [30/886] loss : 0.1604
Epoch [13/50], Iter [40/886] loss : 0.4204
Epoch [13/50], Iter [50/886] loss 

Epoch [14/50], Iter [840/886] loss : 0.7638
Epoch [14/50], Iter [850/886] loss : 0.7501
Epoch [14/50], Iter [860/886] loss : 0.4684
Epoch [14/50], Iter [870/886] loss : 1.2283
Epoch [14/50], Iter [880/886] loss : 0.3165
Average loss: 1.9600 

Epoch    13: reducing learning rate of group 0 to 8.0000e-05.
Learning rate : 8e-05
Epoch [15/50], Iter [10/886] loss : 0.1735
Epoch [15/50], Iter [20/886] loss : 0.2322
Epoch [15/50], Iter [30/886] loss : 1.5127
Epoch [15/50], Iter [40/886] loss : 0.3439
Epoch [15/50], Iter [50/886] loss : 1.3386
Epoch [15/50], Iter [60/886] loss : 0.4027
Epoch [15/50], Iter [70/886] loss : 0.6694
Epoch [15/50], Iter [80/886] loss : 0.1914
Epoch [15/50], Iter [90/886] loss : 0.0761
Epoch [15/50], Iter [100/886] loss : 0.7848
Epoch [15/50], Iter [110/886] loss : 0.6968
Epoch [15/50], Iter [120/886] loss : 0.7523
Epoch [15/50], Iter [130/886] loss : 0.3006
Epoch [15/50], Iter [140/886] loss : 0.5059
Epoch [15/50], Iter [150/886] loss : 0.5231
Epoch [15/50], Iter [1

Epoch [17/50], Iter [60/886] loss : 0.1155
Epoch [17/50], Iter [70/886] loss : 1.1604
Epoch [17/50], Iter [80/886] loss : 0.2326
Epoch [17/50], Iter [90/886] loss : 0.1957
Epoch [17/50], Iter [100/886] loss : 0.1154
Epoch [17/50], Iter [110/886] loss : 0.7361
Epoch [17/50], Iter [120/886] loss : 0.8916
Epoch [17/50], Iter [130/886] loss : 0.0866
Epoch [17/50], Iter [140/886] loss : 0.2793
Epoch [17/50], Iter [150/886] loss : 0.3914
Epoch [17/50], Iter [160/886] loss : 0.1254
Epoch [17/50], Iter [170/886] loss : 0.1479
Epoch [17/50], Iter [180/886] loss : 0.2665
Epoch [17/50], Iter [190/886] loss : 0.1058
Epoch [17/50], Iter [200/886] loss : 0.0528
Epoch [17/50], Iter [210/886] loss : 0.8487
Epoch [17/50], Iter [220/886] loss : 0.5430
Epoch [17/50], Iter [230/886] loss : 0.1372
Epoch [17/50], Iter [240/886] loss : 0.3491
Epoch [17/50], Iter [250/886] loss : 0.2453
Epoch [17/50], Iter [260/886] loss : 0.8737
Epoch [17/50], Iter [270/886] loss : 0.7805
Epoch [17/50], Iter [280/886] loss :

Epoch [19/50], Iter [180/886] loss : 0.6407
Epoch [19/50], Iter [190/886] loss : 0.0708
Epoch [19/50], Iter [200/886] loss : 0.5503
Epoch [19/50], Iter [210/886] loss : 0.1713
Epoch [19/50], Iter [220/886] loss : 0.3752
Epoch [19/50], Iter [230/886] loss : 0.7021
Epoch [19/50], Iter [240/886] loss : 0.2345
Epoch [19/50], Iter [250/886] loss : 0.2097
Epoch [19/50], Iter [260/886] loss : 0.3063
Epoch [19/50], Iter [270/886] loss : 0.2442
Epoch [19/50], Iter [280/886] loss : 0.1623
Epoch [19/50], Iter [290/886] loss : 0.8082
Epoch [19/50], Iter [300/886] loss : 0.9435
Epoch [19/50], Iter [310/886] loss : 0.3337
Epoch [19/50], Iter [320/886] loss : 0.3478
Epoch [19/50], Iter [330/886] loss : 0.6444
Epoch [19/50], Iter [340/886] loss : 0.5516
Epoch [19/50], Iter [350/886] loss : 0.2347
Epoch [19/50], Iter [360/886] loss : 0.2415
Epoch [19/50], Iter [370/886] loss : 0.1827
Epoch [19/50], Iter [380/886] loss : 0.2211
Epoch [19/50], Iter [390/886] loss : 0.2075
Epoch [19/50], Iter [400/886] lo

Epoch [21/50], Iter [300/886] loss : 0.4003
Epoch [21/50], Iter [310/886] loss : 0.5692
Epoch [21/50], Iter [320/886] loss : 0.6243
Epoch [21/50], Iter [330/886] loss : 0.9475
Epoch [21/50], Iter [340/886] loss : 0.1224
Epoch [21/50], Iter [350/886] loss : 0.9799
Epoch [21/50], Iter [360/886] loss : 0.4260
Epoch [21/50], Iter [370/886] loss : 0.4722
Epoch [21/50], Iter [380/886] loss : 0.4393
Epoch [21/50], Iter [390/886] loss : 0.5232
Epoch [21/50], Iter [400/886] loss : 0.7848
Epoch [21/50], Iter [410/886] loss : 0.1403
Epoch [21/50], Iter [420/886] loss : 0.6677
Epoch [21/50], Iter [430/886] loss : 0.7693
Epoch [21/50], Iter [440/886] loss : 0.5271
Epoch [21/50], Iter [450/886] loss : 0.9546
Epoch [21/50], Iter [460/886] loss : 0.2989
Epoch [21/50], Iter [470/886] loss : 0.3545
Epoch [21/50], Iter [480/886] loss : 0.7000
Epoch [21/50], Iter [490/886] loss : 0.5844
Epoch [21/50], Iter [500/886] loss : 2.4854
Epoch [21/50], Iter [510/886] loss : 0.3475
Epoch [21/50], Iter [520/886] lo

plain: Test Accuracy: 0.5207 

# 3s splits without averaging of result
Test Accuracy: 0.5590 

(443, 128, 1287) (443,) (197, 128, 1287) (197,) (290, 128, 1287) (290,)
model_1DCNN(
  (conv0): Sequential(
    (0): Conv1d(128, 32, kernel_size=(8,), stride=(1,))
    (1): BatchNorm1d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (2): ReLU()
    (3): MaxPool1d(kernel_size=8, stride=8, padding=0, dilation=1, ceil_mode=False)
  )
  (conv1): Sequential(
    (0): Conv1d(32, 32, kernel_size=(8,), stride=(1,))
    (1): BatchNorm1d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (2): ReLU()
    (3): MaxPool1d(kernel_size=8, stride=8, padding=0, dilation=1, ceil_mode=False)
  )
  (conv2): Sequential(
    (0): Conv1d(32, 64, kernel_size=(4,), stride=(1,))
    (1): BatchNorm1d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (2): ReLU()
    (3): MaxPool1d(kernel_size=4, stride=4, padding=0, dilation=1, ceil_mode=False)
  )
  (fc0): Linear(in_features=256, out_features=64, bias=True)
  (fc1): Linear(in_features=64, out_features=10, bias=True)
)

In [7]:
path = os.path.abspath('../..')
mel_path = path + '/gtzan_mel'
mel = np.load(mel_path +'/blues/blues.00045.npy')
mel

array([[6.6990161e-01, 3.7650773e-01, 4.2052668e-01, ..., 2.4691856e-01,
        3.1386757e-01, 1.7449795e-01],
       [9.4072378e-01, 7.1411455e-01, 1.0616822e+00, ..., 3.3857328e-01,
        4.2921627e-01, 3.6669487e-01],
       [9.8529458e-01, 7.8986245e-01, 1.3927085e+00, ..., 8.1729567e-01,
        6.6713756e-01, 6.9279540e-01],
       ...,
       [2.5567310e-03, 1.7779057e-04, 1.8117324e-04, ..., 2.5232119e-04,
        2.0861955e-04, 3.5291596e-04],
       [2.4695520e-03, 6.5120585e-05, 5.6098950e-05, ..., 6.9935624e-05,
        5.6794022e-05, 1.2212024e-04],
       [2.4294013e-03, 3.3567940e-05, 3.7775073e-05, ..., 2.7846680e-05,
        2.4686078e-05, 2.9587585e-05]], dtype=float32)

In [8]:
mel.size

164736

In [9]:
mel.shape


(128, 1287)

In [23]:
%run -i 'feature_extraction.py'

ERROR:root:File `u'feature_extraction.py'` not found.


In [24]:
for i in range(1):
    print i

0


In [3]:
import torch
import torch.nn as nn

in_ten = torch.randn(5, 128, 1287)
m = nn.Conv1d(128, 32, kernel_size=8, stride=1, padding=0)
output = m(in_ten)

In [4]:
output.shape

torch.Size([5, 32, 1280])

In [5]:
m1 = nn.BatchNorm1d(32)
output2 = m1(output)
print(output2.shape)

torch.Size([5, 32, 1280])


In [6]:
m2 = nn.ReLU()
output3 = m2(output2)
print(output3.shape)
m3 = nn.MaxPool1d(8, stride=8)
output4 = m3(output3)
print(output4.shape)

torch.Size([5, 32, 1280])
torch.Size([5, 32, 160])
